### PR TITLE
Release notes

### PR DESCRIPTION
**Pull request by sanastas**
Adding the release notes to be updated from now on

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: release-notes.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+======================================================
```
Comment (liran-funaro) [inline release-notes.md:1]: What is this line?

File: release-notes.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+## OAK 0.2 Release Notes (June 01, 2020)
+>Compared to Oak 0.1.5 that was published on May 14, 2019
```
Comment (liran-funaro) [inline release-notes.md:4]: Maybe instead just add below a title for the last release with one sentence of what you remember it was about.
And remove this line from here. For example:

## v0.1.5 (May 14, 2019)
 * API changes

File: release-notes.md (hunk 1 above)
Comment (liran-funaro) [reply release-notes.md:1]: Maybe we should have a title here instead:
# Oak Release Notes

File: release-notes.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ 06. Refactoring for future OakHash: EntrySet class is introduced to encapsulate (1) the entries to integers mapping and (2) managing (allocate/read/write/delete/release) individual key-value entries. Helps in future OakHash coding.
+ 07. No internal objects creation: Eliminate ephemeral objects creation in time of lookup. It greatly reduces the work to be done by GC and improves performance.
+ 08. Safe/Unsafe API: Create a possibility to work with OakBuffers in a fully safe/protected mode, so no wrong memory can be accessed/written erroneously. Safe mode requires some performance degradation, thus an alternative unsafe mode is introduced, providing greater performance and relying on the correct user actions for the security.
+ 09. Improving benchmarks code: Reducing the time it takes to run all benchmarks. Improving the results readability. Improving the benchmark scripts user experience.
+ 10. Correctness bug fixes: Various bugs fixes: in non-zero-copy API, internal memory access, rebalance process, etc.
```
Comment (liran-funaro) [inline release-notes.md:16]: You repeated "bug fixes" twice. The second "Various bugs fixes:" is redundant.
"etc." is incorrect here.
I suggest "Correctness bug fixes (partial list): ..."
The same below.

File: release-notes.md (hunk 1 above)
Comment (sanastas) [reply release-notes.md:1]: This line I thought to use for the future separation between releases. Like the future release notes should come above this line. I wanted to put it there so future updater won't forget to separate between releases. But if it is confusing I can remove it.

File: release-notes.md (hunk 2 above)
Comment (sanastas) [reply release-notes.md:4]: OK, I will do it.

File: release-notes.md (hunk 3 above)
Comment (sanastas) [reply release-notes.md:14]: OK, fixed

**Pull request merged**